docs/hapi.asciidoc

```diff
@@ -0,0 +1,188 @@
+[[hapi]]
+== Getting started with hapi
+
+Getting Elastic APM set up for your hapi app is easy,
+and there are various ways you can tweak it to fit to your needs.
+
+Follow the guide below and refer to the <<api,API documentation>> for all the advanced stuff.
+
+[[hapi-installation]]
+=== Installation
+
+Add the `elastic-apm` module as a dependency to your application:
+
+[source,bash]
+----
+npm install elastic-apm --save
+----
+
+[[hapi-initialization]]
+=== Initialization
+
+It's important that the agent is started before you require *any* other modules in your Node.js application - i.e. before `hapi`, `http`, etc.
+
+This means that you should probably require and start it in your applications main file (usually `index.js`, `server.js` or `app.js`).
+
+Here's a simple hapi example with the Elastic APM agent installed:
+
+[source,js]
+----
+// Add this to the VERY top of the first file loaded in your app
+var apm = require('elastic-apm').start({
+  // Set required app name (allowed characters: a-z, A-Z, 0-9, -, _, and space)
+  appName: '',
+
+  // Use if APM Server requires a token
+  secretToken: '',
+
+  // Set custom APM Server URL (default: http://localhost:8200)
+  serverUrl: '',
+})
+
+var Hapi = require('hapi')
+
+var server = new Hapi.Server()
+
+server.connection({
+  host: 'localhost',
+  port: 8000
+})
+
+server.route({
+  method: 'GET',
+  path: '/hello',
+  handler: function (request, reply) {
+    return reply('hello world')
+  }
+})
+
+server.start()
+----
+
+The agent will now monitor the performance of your hapi application and record any uncaught exceptions.
+
+[[hapi-advanced-configuration]]
+==== Advanced configuration
+
+In the above example we initialize the agent by calling the <<apm-start,`start()`>> function.
+This function takes an optional options object used to configure the agent.
+Any option not supplied via the options object can instead be configured using environment variables.
+So if you prefer, you can set the same configuration options using environment variables:
+
+[source,bash]
+----
+ELASTIC_APM_APP_NAME=<app name>
+ELASTIC_APM_SECRET_TOKEN=<token>
+ELASTIC_APM_SERVER_URL=<server url>
+----
+
+And then just start the agent like so:
+
+[source,js]
+----
+// Start the agent before any thing else in your app
+var apm = require('elastic-apm').start()
+----
+
+See all possible ways to configure the agent <<configuring-the-agent,in the API documentation>>.
+
+[[hapi-full-documentation]]
+==== Full documentation
+
+* <<setup,Setup and Configuration>>
+* <<api,API Reference>>
+
+[[hapi-performance-monitoring]]
+=== Performance monitoring
+
+Elastic APM automatically measures the performance of your hapi application.
+It records traces for database queries,
+external HTTP requests,
+and other slow operations that happens during requests to your hapi app.
+
+By default the agent will trace <<compatibility,the most common modules>>.
```

Comment:
did you mean to link to compatibility here?

Reply:
Yes, the `compatibility` page lists all the modules we currently support